### DISCUSSION TITLE
Serializability / Persistence

### DISCUSSION THREAD

**Discussion by aedenj** [nsk90/kstatemachine#97] [General]
I have a need to serialize state for application failure and recovery use cases. I read the section on persistence and I'm wondering whether there is a feature being worked on for persisting & restoring the configuration and if not what would that effort be to do something like that? The event recording option while possible would be very memory intensive in my case.

**Comment by nsk90** [comment]
Hi, now I am working on event recording use case. State serialization is not even in progress yet.   It is pretty complicated and there are many things to serialize. 

You can specify StateMachine features that you use, so I can at least know were to start from.

**Reply by aedenj** [reply]
I'm not entirely sure how to answer this so I'm just going to provide a list based on your docs.

* All the different types of transitions
* Various state types, DefaultState, Final state etc.. DataState, DataEvent etc..
* History State

Does that help?

**Reply by nsk90** [reply]
Yes sure, thank you
